### ISSUE TITLE
[BUG] type checker gives false positive for WithTransform + ContainElement

### ISSUE THREAD

**Issue by LittleFox94** [nunnatsa/ginkgolinter#115]
**Describe the bug**
The new type check linting gives a false positive when using a `WithTransform` to transform e.g. an `[]int` into an `[]string` together with `ContainElement(string)`:

`ginkgo-linter: use ContainElement with different types: Comparing []string with string; either change the expected value type if possible, or use the BeEquivalentTo() matcher, instead of ContainElement() (ginkgolinter)`

**To Reproduce**
```go
Expect([]int{42, 23}).Should(WithTransform(func(v []int) []string {
	ret := make([]string, 0, len(v))
	for _, i := range v {
		ret = append(ret, fmt.Sprintf("%v", i))
	}
	return ret
}, ContainElement("42")))
```

**Expected behavior**
Using `ContainElement(string)` on the output of a `WithTransform` function that returns a `[]string` is correct and the linter should not report an error for that.

**Environment:**
 - OS: Linux
 - Version v0.14.0

**Additional context**

Originally spotted in our [CI here](https://github.com/anexia-it/go-anxcloud/actions/runs/6649258043/job/18067504607?pr=307#step:4:193), the problematic code being [here](https://github.com/anexia-it/go-anxcloud/blob/00237d6e747c18fc2eb2b05c5a17a0145b99caf1/pkg/clouddns/zone/zone_test.go#L154-L160).

**Comment by nunnatsa** [comment]
Thanks for letting me know, @LittleFox94!

**Comment by nunnatsa** [comment]
@LittleFox94 - would you like to review the fix? https://github.com/nunnatsa/ginkgolinter/pull/116

**Comment by LittleFox94** [comment]
Heya, sorry for my late reply .. seems to be ok now - thank you :)

**Comment by nunnatsa** [comment]
The fix is included in golangci-lint version v1.55.2
